boxsdk/object/base_object.py

```diff
@@ -155,8 +155,24 @@ def delete(self, params=None, headers=None):
         return box_response.ok
 
     def __eq__(self, other):
-        """Equality as determined by object id"""
-        return self._object_id == other.object_id
+        """Equality as determined by object id and type"""
+        if (other is None):
+            return False
+        if not isinstance(other, BaseObject):
+            return False
+
+        # Two objects are considered the same if they have the same address in the API
+        return self.get_url() == other.get_url()
```

Comment:
Cool approach. I hadn't thought of doing it this way, but this makes a lot of sense.

One thing to note about this approach (which is probably, but not necessarily, an upside) is that two objects will compare as unequal if they are the same except for the version of the API that they are pointing at.